Quote tab completion items with PS special chars.

rkeithhill opened the pull request:
Fix #293

rkeithhill commented:
If we merge this for 0.7.0, don't forget to update CHANGELOG.md.

dahlbyk commented on src/GitTabExpansion.ps1:
```diff
@@ -51,6 +51,15 @@ catch {
 
+filter quoteStringWithSpecialChars {
+    if ($_ -and ($_ -match '\s+|#|@|\$|;|\{|\}|\(|\)')) {
+        "'" + $_ + "'"
```
1. Why not `"'$_'"`?
2. We should also handle values that include `'`.

dahlbyk commented on src/GitTabExpansion.ps1:
```diff
@@ -51,6 +51,15 @@ catch {
 }
 
+filter quoteStringWithSpecialChars {
+    if ($_ -and ($_ -match '\s+|#|@|\$|;|\{|\}|\(|\)')) {
```
Missing `,`

dahlbyk commented on test/TabExpansion.Tests.ps1:
```diff
@@ -137,16 +137,70 @@ Describe 'TabExpansion Tests' {
     }
+
+    Context 'PowerShell Special Chars Tests' {
+        BeforeAll {
```
This is bound to be a pretty common pattern. Is it possible to write a Pester helper that would encapsulate a `BeforeAll`/`AfterAll` pairing that initializes and cleans up a Git repo, with its path left in `$repoPath`? Something like:

```powershell
Context 'Awesome test' {
    TempRepo {
        # Set up the repo
    }
    It 'Uses $repoPath' {
        # ...
    }
}
```

rkeithhill commented:
I think I've addressed all of the issues except the last one the Pester helper.  That said, I did move common code into reusable functions (in Shared.ps1).
